pymc3/math.py

```diff
@@ -18,10 +18,84 @@
 import scipy.sparse
 from scipy.linalg import block_diag as scipy_block_diag
 from pymc3.theanof import floatX, largest_common_dtype, ix_
+from functools import reduce, partial
 
 # pylint: enable=unused-import
 
 
+def kronecker(*Ks):
+    """Return the Kronecker product of arguments:
+            K_1 \otimes K_2 \otimes ... \otimes K_D
+
+    Parameters
+    ----------
+    Ks: 2D array-like
+    """
+    return reduce(tt.slinalg.kron, Ks)
```

Comment:
Isnt `reduce` only available in python 3?

Reply:
It looks it's available [in Python 2](https://docs.python.org/2/library/functools.html#functools.reduce) as well.

Reply:
Does it return list or generator in Python 3?

Reply:
Neither. It returns a theano tensor object that is the result of the nested evaluations `kron(kron(...(kron(Ks[0], Ks[1]), Ks[2]), ...), Ks[d])`, that is, the Kronecker product of all the arrays in order.